### PR TITLE
feat(iOS): screensharing extension swift implementation

### PR DESCRIPTION
**Pull request by alexbumbu**

### CODE REVIEW

File: ios/app/JitsiMeetBroadcast Extension/DarwinNotificationCenter.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//
+//  DarwinNotificationCenter.swift
+//  Broadcast Extension
+//
+//  Created by Alex-Dan Bumbu on 23/03/2021.
+//  Copyright © 2021 Atlassian Inc. All rights reserved.
```
Comment (saghul) [inline ios/app/JitsiMeetBroadcast Extension/DarwinNotificationCenter.swift:6]: Heh nope. Let's drop all copyright headers here, they are not needed anyway.

File: ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+//  Broadcast Extension
+//
+//  Created by Alex-Dan Bumbu on 22/03/2021.
+//  Copyright © 2021 Atlassian Inc. All rights reserved.
+//
+
+import ReplayKit
+
+private enum Constants {
+    static let appGroupIdentifier = "group.org.jitsi.meet.appgroup"
```
Comment (saghul) [inline ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift:12]: Maybe put a comment mentioning this needs to be adapted for each app?

File: ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          clientConnection = connection
+          setupConnection()
+          
+          uploader = SampleUploader(connection: connection)
+        }
+    }
+
+    override func broadcastStarted(withSetupInfo setupInfo: [String: NSObject]?) {
+        // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+        print("broadcast started")
```
Comment (saghul) [inline ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift:40]: Can we use our logger here?

File: ios/app/JitsiMeetBroadcast Extension/SampleUploader.swift [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+private extension SampleUploader {
+    
+    func setupConnection() {
+        connection.didOpen = { [weak self] in
+            self?.isReady = true
+        }
+        connection.streamHasSpaceAvailable = { [weak self] in
+            self?.serialQueue.async {
+                self?.isReady = !(self?.sendData() ?? true)
```
Comment (saghul) [inline ios/app/JitsiMeetBroadcast Extension/SampleUploader.swift:55]: We are setting this in the serial queue but reading it from a different thread. Maybe we should check inside sendData?

**Review by saghul — CHANGES_REQUESTED** [review]
Left some comments, PTAL. General note: please leave the extension where it was, don't change the directory name.

File: ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift (hunk 2 above)
Comment (alexbumbu) [reply ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift]: done

File: ios/app/JitsiMeetBroadcast Extension/SampleUploader.swift (hunk 4 above)
Comment (alexbumbu) [reply ios/app/JitsiMeetBroadcast Extension/SampleUploader.swift]: fixed it using atomic properties. This way we keep only the network communication in the background thread.

File: ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift (hunk 3 above)
Comment (alexbumbu) [reply ios/app/JitsiMeetBroadcast Extension/SampleHandler.swift]: cocoa lumberjack instances are not shared between the app and the extension. If we want this we need to implement a custom mechanism for forwarding messages from the extension to the app.

File: ios/app/broadcast-extension/SampleUploader.swift [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        
+        guard let messageData = bufferData else {
+            print("corrupted image buffer")
+            return nil
+        }
+              
+        let httpResponse = CFHTTPMessageCreateResponse(nil, 200, nil, kCFHTTPVersion1_1).takeRetainedValue()
+        CFHTTPMessageSetHeaderFieldValue(httpResponse, "Content-Length" as CFString, String(messageData.count) as CFString)
+        CFHTTPMessageSetHeaderFieldValue(httpResponse, "Buffer-Width" as CFString, String(width) as CFString)
+        CFHTTPMessageSetHeaderFieldValue(httpResponse, "Buffer-Height" as CFString, String(height) as CFString)
```
Comment (saghul) [inline ios/app/broadcast-extension/SampleUploader.swift:130]: Please port the orientation code from the ObjC extension.

**Review by saghul — CHANGES_REQUESTED** [review]
One final comment, LGTM otherwise.

File: ios/app/broadcast-extension/SampleUploader.swift (hunk 5 above)
Comment (alexbumbu) [reply ios/app/broadcast-extension/SampleUploader.swift:131]: done

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Excellent work @alexbumbu !

**Pull request merged**